logchain/src/segment_block_index/segment_block_index_errors.rs

```diff
@@ -0,0 +1,12 @@
+use util::error::{Error, ErrorType};
+
+pub fn block_index_from_buffer_insufficient_buffer_size(buffer_size: usize) -> Error {
+    Error::new(
+        ErrorType::Happens,
+        "block_index_from_buffer_insufficient_buffer_size",
+        Some(format!(
+            "Insufficient buffer size:\n\t Buffer size: {}",
+            buffer_size
+        )),
```

Comment:
use `ErrorType::Critical`

This error occurs in case of logical error if storage file is corrupt

Reply:
```rs
    Error::new(
        ErrorType::Critical,
        "block_index_from_buffer_insufficient_buffer_size",
        Some(format!(
            "Possible logical error or storage corrupt: Insufficient buffer size:\n\t Buffer size: {}",
            buffer_size
        )),
    )
```